src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -313,7 +383,24 @@ public static String getContainerSettingJson(int startTagIndex, int numToProcces
     }
     for (int i = startTagIndex; i < endTagIndex; i++) {
       if (TagInfoManager.getTagInfoList().get(i) != null) {
-        String tagName = ((TagInfo) TagInfoManager.getTagInfoList().get(i)).getName();
+        TagInfo currentTag = ((TagInfo) TagInfoManager.getTagInfoList()).get(i);
+        String tagName = currentTag.getName();
+
+        if (currentTag.getType() == TagType.FLOAT) {
+          typeID = "HMS-number-type-" + getFlexyName();
+        } else if (currentTag.getType() == TagType.INTEGER) {
+          typeID = "HMS-number-type-" + getFlexyName();
+        } else if (currentTag.getType() == TagType.BOOLEAN) {
+          typeID = "HMS-boolean-type-" + getFlexyName();
+        } else if (currentTag.getType() == TagType.DWORD) {
+          typeID = "HMS-number-type-" + getFlexyName();
+        } else if (currentTag.getType() == TagType.STRING) {
+          typeID = "HMS-string-type-" + getFlexyName();
+        } else {
+          Logger.LOG_SERIOUS("Unsupported tag type was selected for tag " + tagName);
```

Comment:
Any issue with typeID not being initialized?

Reply:
I've never seen it happen. Every type gets in there even if they dont get used so they exist for containers to use. There's only 3. More likely it will come back with a duplicate type creation failed but the response code for that is still 0